### PR TITLE
Add Local Fonts API article

### LINKED ISSUE #2839: content: New Font APIs [2020-06-30]
* [draft](https://docs.google.com/document/d/1eECkz-Vt94Kqoq_9H1f4rPwFQNyiehZU5W1P0Ta0L7c/edit)

### PR DESCRIPTION
**Pull request by tomayac**
Fixes #2839

Changes proposed in this pull request:

- Add the article

### CODE REVIEW

**Comment by tomayac** [comment]
Google Doc: https://docs.google.com/document/d/1eECkz-Vt94Kqoq_9H1f4rPwFQNyiehZU5W1P0Ta0L7c/edit.

File: src/site/content/en/blog/local-fonts/index.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+The Chrome team has designed and implemented the Local Font Access API using the core principles defined in [Controlling Access to Powerful Web Platform Features][powerful-apis], including user control, transparency, and ergonomics.
+
+### User control
+
+Access to a user's fonts is fully under their control and will not be allowed unless the `"local-fonts"` permission, as listed in the
+[permission registry](https://w3c.github.io/permissions/#permission-registry), is granted.
+
+### Transparency
+
+The fact whether a site has been granted access to the user's local fonts will be visible in the *site information* sheet.
```
Comment (jpmedley) [inline src/site/content/en/blog/local-fonts/index.md:291]:
```suggestion
Whether a site has been granted access to the user's local fonts will be visible in the *site information* sheet.
```

Comment (jpmedley) [inline src/site/content/en/blog/local-fonts/index.md:291]: What's a site information sheet? Maybe add a link.

Comment (jpmedley) [reply src/site/content/en/blog/local-fonts/index.md:291]: Is that the thing that pops up when you click the lock icon in the address bar? Maybe show a picture instead.

File: src/site/content/en/blog/local-fonts/index.md [hunk 2]
```diff
@@ -0,0 +1,348 @@
+---
+layout: post
+title: Create advanced typography content with local fonts
```
Comment (tomayac) [inline src/site/content/en/blog/local-fonts/index.md:3]:
```suggestion
title: Use advanced typography with local fonts
```

[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/local-fonts/index.md:3]: That's a good suggestion. Thanks!

File: src/site/content/en/blog/local-fonts/index.md [hunk 3]
```diff
@@ -0,0 +1,348 @@
...
+
+## The Local Font Access API
+
+The beginning of this article may have put you in a negative mood. Can we really not have nice things? Fret not. We think we can, and maybe [everything is not hopeless](http://hyperboleandahalf.blogspot.com/2013/05/depression-part-two.html#Blog1:~:text=like-,hopeless). But first, let me answer a question that you might be asking yourself.
+
+### Why do we need the Local Font Access API when there are web fonts?
+
+Professional-quality design and graphics tools have historically been difficult to deliver on the web. One stumbling block has been an inability to access and use the full variety of professionally constructed and hinted fonts that designers have locally installed. Web fonts enable some publishing use-cases, but fail to enable programmatic access to the vector glyph shapes and a set other font internals, and the font binary data.
```
[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/local-fonts/index.md:104]: It's indeed meant as "access to (the vector glyph shapes and a set other font internals), and (the font binary data)".

Comment (tomayac) [inline src/site/content/en/blog/local-fonts/index.md:104]:
```suggestion
Professional-quality design and graphics tools have historically been difficult to deliver on the web. One stumbling block has been an inability to access and use the full variety of professionally constructed and hinted fonts that designers have locally installed. Web fonts enable some publishing use-cases, but fail to enable programmatic access to the vector glyph shapes and a set of other font internals, and the font binary data.
```

File: src/site/content/en/blog/local-fonts/index.md [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+For example, anonymity-focused browsers may choose to only provide
+a set of default fonts built into the browser.
+Similarly, browsers are not required to provide table data exactly as it appears on disk.
+
+Wherever possible, the Local Font Access API is designed to only expose exactly the information
+needed to enable the mentioned use cases.
+System APIs may produce a list of installed fonts not in a random or a sorted order,
+but in the order of font installation.
+Returning exactly the list of installed fonts given by such a system API
+can expose additional entropy bits, and use cases we want to enable
```
Comment (tomayac) [inline src/site/content/en/blog/local-fonts/index.md:276]:
```suggestion
can expose additional data that may be used for fingerprinting, and use cases we want to enable
```

[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/local-fonts/index.md]: Changed. Thanks!

File: src/site/content/en/blog/local-fonts/index.md [hunk 5]
```diff
@@ -0,0 +1,348 @@
...
+The Chrome team has designed and implemented the Local Font Access API using the core principles defined in [Controlling Access to Powerful Web Platform Features][powerful-apis], including user control, transparency, and ergonomics.
+
+### User control
+
+Access to a user's fonts is fully under their control and will not be allowed unless the `"local-fonts"` permission, as listed in the
+[permission registry](https://w3c.github.io/permissions/#permission-registry), is granted.
+
+### Transparency
+
+Whether a site has been granted access to the user's local fonts will be visible in the *site information* sheet.
```
Comment (tomayac) [inline src/site/content/en/blog/local-fonts/index.md:291]:
```suggestion
Whether a site has been granted access to the user's local fonts will be visible in the
[site information sheet](https://support.google.com/chrome/answer/114662?hl=en&co=GENIE.Platform=Desktop).
```

File: src/site/content/en/blog/local-fonts/index.md (hunk 1 above)
Comment (tomayac) [reply src/site/content/en/blog/local-fonts/index.md]: It is, yes. For clarity, I have added a link: https://support.google.com/chrome/answer/114662?hl=en&co=GENIE.Platform=Desktop.

File: src/site/content/en/blog/local-fonts/index.md [hunk 6]
```diff
@@ -0,0 +1,349 @@
...
+  - blog
+  - fonts
+  - capabilities
+authors:
+  - thomassteiner
+description: >
+  The Local Fonts API enumerates the user's installed local
+  fonts and provides low-level access to the various
+  TrueType/OpenType tables.
+date: 2020-08-03
```
Comment (tomayac) [inline src/site/content/en/blog/local-fonts/index.md:16]:
```suggestion
date: 2020-08-12
```

File: src/site/content/en/blog/local-fonts/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jpmedley) [reply src/site/content/en/blog/local-fonts/index.md]: We need make this less confusing. What does 'a set of other font internals' mean? Is that the set of other font internals for the vector glyphs?

File: src/site/content/en/blog/local-fonts/index.md [hunk 7]
```diff
@@ -0,0 +1,349 @@
...
+
+## The Local Font Access API
+
+The beginning of this article may have put you in a negative mood. Can we really not have nice things? Fret not. We think we can, and maybe [everything is not hopeless](http://hyperboleandahalf.blogspot.com/2013/05/depression-part-two.html#Blog1:~:text=like-,hopeless). But first, let me answer a question that you might be asking yourself.
+
+### Why do we need the Local Font Access API when there are web fonts?
+
+Professional-quality design and graphics tools have historically been difficult to deliver on the web. One stumbling block has been an inability to access and use the full variety of professionally constructed and hinted fonts that designers have locally installed. Web fonts enable some publishing use-cases, but fail to enable programmatic access to the vector glyph shapes and a set of other font internals, and the font binary data.
```
Comment (tomayac) [inline src/site/content/en/blog/local-fonts/index.md:104]:
```suggestion
Professional-quality design and graphics tools have historically been difficult to deliver on the web.
One stumbling block has been an inability to access and use the full variety of
professionally constructed and hinted fonts that designers have locally installed.
Web fonts enable some publishing use-cases, but fail to enable programmatic access to
the vector glyph shapes and font tables used by rasterizers to render the glyph outlines.
There is likewise no way to access a web font's binary data.
```

File: src/site/content/en/blog/local-fonts/index.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (tomayac) [reply src/site/content/en/blog/local-fonts/index.md]: I gave it [another shot](https://github.com/GoogleChrome/web.dev/pull/3615/commits/c2e6c5fd51b7aa3c4ae7e5d4ca3aafe6a98b4d8b).

**Comment by kaycebasques** [comment]
@tomayac just making sure you know that @jpmedley has merge access now, so once he LGTM's your PRs you should be able to merge now

File: src/site/content/en/blog/local-fonts/index.md [hunk 8]
```diff
@@ -0,0 +1,354 @@
...
+  - blog
+  - fonts
+  - capabilities
+authors:
+  - thomassteiner
+description: >
+  The Local Fonts API enumerates the user's installed local
+  fonts and provides low-level access to the various
+  TrueType/OpenType tables.
+date: 2020-08-12
```
Comment (tomayac) [inline src/site/content/en/blog/local-fonts/index.md:16]:
```suggestion
date: 2020-08-18
```

**Comment by tomayac** [comment]
@kaycebasques This is good to go live. Since I have removed the "Draft PR" status, it now requires a new LGTM from someone. Thanks!

**Comment by jpmedley** [comment]
@robdodson This PR has been waiting on the List and test since yesterday afternoon. There's nothing I can click to see what's holding it up. Can you help us out?

**Comment by robdodson** [comment]
yeah let me take a look. We made some changes to that action recently and I'm wondering if PRs created before the change will need to rebase. I can rebase it and see

**Comment by robdodson** [comment]
It looks like it's running now after the rebase. So I think if a PR was created more than one week ago we will need to rebase them to get this check to run.

File: src/site/content/en/blog/local-fonts/index.md [hunk 9]
```diff
@@ -0,0 +1,354 @@
...
+  - blog
+  - fonts
+  - capabilities
+authors:
+  - thomassteiner
+description: >
+  The Local Fonts API enumerates the user's installed local
+  fonts and provides low-level access to the various
+  TrueType/OpenType tables.
+date: 2020-08-18
```
Comment (tomayac) [inline src/site/content/en/blog/local-fonts/index.md:16]:
```suggestion
date: 2020-08-19
```

File: src/site/content/en/blog/local-fonts/index.md [hunk 10]
```diff
@@ -0,0 +1,354 @@
...
+
+## Helpful links
+
+- [Explainer][explainer]
+- [Spec draft][spec]
+- [Chromium bug for font enumeration][cr-bug-enum]
+- [Chromium bug for font table access][cr-bug-table]
+- [ChromeStatus entry][cr-status]
+- [GitHub repo][issues]
+- [Tag review][tag-review]
```
Comment (tomayac) [inline src/site/content/en/blog/local-fonts/index.md:331]:
```suggestion
- [Tag review][tag-review]
- [Mozilla standards position][mozilla]
```

File: src/site/content/en/blog/local-fonts/index.md [hunk 11]
```diff
@@ -0,0 +1,354 @@
...
+
+[tag-review]: https://github.com/w3ctag/design-reviews/issues/399
+[explainer]: https://github.com/WICG/local-font-access
+[spec]: https://wicg.github.io/local-font-access/
+[powerful-apis]: https://chromium.googlesource.com/chromium/src/+/lkgr/docs/security/permissions-for-powerful-web-platform-features.md
+[issues]: https://github.com/WICG/local-font-access/issues
+[cr-status]: https://chromestatus.com/feature/6234451761692672
+[cr-bug-enum]: https://bugs.chromium.org/p/chromium/issues/detail?id=535764
+[cr-bug-table]: https://crbug.com/982054
+[cr-dev-twitter]: https://twitter.com/ChromiumDev
```
Comment (tomayac) [inline src/site/content/en/blog/local-fonts/index.md:354]:
```suggestion
[cr-dev-twitter]: https://twitter.com/ChromiumDev
[mozilla]: https://github.com/mozilla/standards-positions/issues/401
```

**Pull request merged**